accepted/single-file/design.md

```diff
@@ -66,225 +57,209 @@ Publishing to a single file can be triggered by adding the following property to
 
-```cs
-namespace System.Runtime.Loader
-{
-    public partial class Bundle
-    {
-        // Check whether an app is running from a single-file bundle
-        public static bool IsBundle(Assembly assembly);
-        
-        // Get the location where contents of the bundle are extracted
-        public static string GetContentRoot(Assembly assembly);
- 
-        // Open a file embedded in the bundle built for the specified assembly 
-        public static System.IO.Stream GetFileStream(Assembly assembly, string name);
-    }
-}
-```
+The files typically published for self-contained apps on Windows are:
+
+* The `App.Exe`-- StaticHost along with assemblies and configuration files bundled as a single-file app.
+* `coreclr.dll`(runtime), `clrjit.dll` (JIT compiler), and `clrcompression.dll` (native counterpart of BCL for certain compression algorithms).
+    * It may be possible to link these DLLs together as CoreCLR.dll, but this work is not prioritized, in deference to building the full super-host once debugging framework supports it.
+* mscordaccore.dll (to enable Watson dumps)
+
+Certain additional binaries may be optionally included with the app to enable debugging scenarios. 
+The work to not require these binaries as separate files on disk alongside the app is ongoing.
+
+* For F5 debugging on VS and VS-Core: `mscordbi.dll`, `libmscordbi.so`
+* For Linux mini-dumps: `createdump`, `libmscordaccore.so` 
+* For ETW / LTTng: `clretwrc.dll`, `libcoreclrtraceptprovider.so` 
+* For exception stack trace source information, error string resources on Windows: `Microsoft.DiaSymReader.Native.amd64.dll`, `mscorrc.debug.dll`, `mscorrc.dll`
+
+When targeting `win7` platform, several additional DLLs are necessary (`api-*.dll`) to handle API compatibility. These files must be alongside the AppHost for the app to even start execution.
+Therefore, we propose that targeting `win7-*` should not be supported when publishing apps as a single-file.
+
+## The Runtime
+
+### Dependency Resolution
+
+The Host includes bundled assemblies as relative paths in `TRUSTED_PLATFORM_ASSEMBLIES`. When such a path is encountered, the runtime probes the bundle using the `bundle_probe` callback to obtain the location of the assembly within the single-file bundle. It then proceeds to memory-map and load the assembly as described [below](#PEImage-loader).
 
-We can also provide an abstraction that abstracts away the physical location of a file (bundle or disk). For example, add a variant of  `GetFileStream` API that looks for a file in the bundle, and if not found, falls back to disk-lookup. However such abstractions are also easy to build outside of the .Net Framework.
+### PEImage loader
 
-### Existing API
+* IL assemblies are loaded directly from the bundle.
+    * The portion of the single-file bundle containing the required assembly is memory mapped, and the contents are appropriately interpreted by the runtime.
```

Comment:
The entire bundle file can be quite large, and on 32-bit systems there might be not enough virtual address to do this. There should be a fallback.

Reply:
Sure @lpereira; I'll remove the comment about mappint the bundle -- which is an implementation detail.  We only need to map the headers at startup.

Reply:
> 32-bit systems there might be not enough virtual address to do this

The bundle sizes large enough to hit this problem should be a explicit non-goal.